src/utils/utils.ts

```diff
@@ -13,25 +13,27 @@ import input from "../input.json";
 import { PrimitiveType } from "../types/AttributeType";
 
 export const getOrCreateDomainModel = async (
-  model: IModel
+  model: IModel,
+  moduleName: string
 ): Promise<domainmodels.DomainModel> => {
-  return await getDomainModelInterface(model).load();
+  return await getDomainModelInterface(model, moduleName).load();
 };
 
-const getDomainModelInterface = (model: IModel): domainmodels.IDomainModel => {
+const getDomainModelInterface = (model: IModel, moduleName: string): domainmodels.IDomainModel => {
   const existingDomainModelInterface = model
     .allDomainModels()
-    .filter((dm) => dm.containerAsModule.name === input.Module)[0];
+    .filter((dm) => dm.containerAsModule.name === moduleName)[0];
   if (existingDomainModelInterface) return existingDomainModelInterface;
```

Comment:
You can also use ternary operations in return statements:

`return existingDomainModelInterface ? existingDomainModelInterface : createDomainModelInterface(model, moduleName);`

Although, also here, I think you can use the following:

`return existingDomainModelInterface || createDomainModelInterface(model, moduleName);`